src/tools/illink/test/Mono.Linker.Tests/TestCasesRunner/ResultChecker.cs

```diff
@@ -308,7 +279,8 @@ protected virtual void AdditionalChecking (LinkedTestCaseResult linkResult, Asse
 
 		protected virtual void InitialChecking (LinkedTestCaseResult linkResult, AssemblyDefinition original, AssemblyDefinition linked)
 		{
-			_peVerifier.Check (linkResult, original);
+			CreateILChecker ().Check(linkResult, original);
+			ValidateTypeRefsHaveValidAssemblyRefs (linked);
```

Comment:
I wasn't entirely sure what to do about `ValidateTypeRefsHaveValidAssemblyRefs `.  I did find it a bit weird that it was called in two places.  And it might be nice if `VerifyILOfOtherAssemblies` also checked winmd files.

As for moving `ValidateTypeRefsHaveValidAssemblyRefs` into `ILChecker`, I looked at it briefly and ran into `VerifyILOfOtherAssemblies` needing to call `ResolveLinkedAssembly` which isn't currently available to `ILChecker` (not that it couldn't be made available).  I think for now I'd prefer to keep this PR contained so I can get it landed.  As for the future, I think it would be OK to move `ValidateTypeRefsHaveValidAssemblyRefs` in `ILChecker`